### PR TITLE
Never use exponential if radix not 10 (Fix #3739, #3740)

### LINKED ISSUE #3739: Follow-up #2751: For Number#toString with a radix other than 10, never switch to exponential display
Follow-up to #2751 (PR #3635)

Compatibility: align with other engines' implementation here.

See @littledan's comment https://github.com/Microsoft/ChakraCore/issues/2751#issuecomment-326902301

/cc @xiaoyinl @littledan @bterlson

### LINKED ISSUE #3740: Follow-up #2751: Number#toString: add tests on the edge cases between written-out and exponential notation
Follow-up to #2751 (PR #3635)

See comments on PR: https://github.com/Microsoft/ChakraCore/pull/3635

/cc @xiaoyinl

### PR DESCRIPTION
**Pull request by xiaoyinl**
Other engines (SpiderMonkey, JSC and V8) never use exponential display for num.toString(radix) if radix is not 10, no matter how large num is. This PR updates the code and the test cases to match that behavior.

This PR fixes #3739 and fixes #3740.

See also: #2751, #3635.

/cc @dilijev @littledan @bterlson

### CODE REVIEW

File: test/Number/toString.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ runTest('new Number("444" + "123.789123456789875436")');
 runTest('new Number(-444123.78963636363636363636)');
 runTest('new Number(0)');
 runTest('1e21');
+runTest('1.1e21');
```
Comment (dilijev) [inline test/Number/toString.js]: Maybe also add tests for `1e21+1` and `1e21-1` (integers one before and after the boundary) and compare output with other engines. Maybe `1e21+0.1` and `1e21-0.1` also?

Comment (xiaoyinl) [reply test/Number/toString.js]: These numbers all equal to `1e21` due to the loss of precision. How about adding `0.9999999999999999e21` and `1.0000000000000001e21`?

**Comment by xiaoyinl** [comment]
@dotnet-bot test Ubuntu static_ubuntu_linux_test please

File: test/Number/toString.js (hunk 1 above)
Comment (dilijev) [reply test/Number/toString.js]: Yeah, that sounds good. :)

**Comment by jackhorton** [comment]
Any updates on this? I believe it is blocking https://github.com/nodejs/node-chakracore/issues/390.

**Comment by MikeHolman** [comment]
So just for fun I did some archeology to see why we are doing this, and turns out this code goes all the way back to at least 2002 (when it was copied between source trees). I don't think anyone has considered this behavior in a very long time.

Seems to me like a fine change to get compat with the other browsers.

**Comment by dilijev** [comment]
@MikeHolman can you adopt and merge this PR?

**Comment by dilijev** [comment]
Spoke with @MikeHolman - I'll test this one internally and then merge it.

**Comment by dilijev** [comment]
Note: merging to release/1.7 to unblock the node-chakracore scenario: https://github.com/Microsoft/ChakraCore/pull/3742#issuecomment-335887549

**Pull request merged**